Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoASA.yml

```diff
@@ -0,0 +1,369 @@
+                | parse Message with * "access-list " * " " * " " * " " SrcInterfaceName "/" * ") -> " DstInterfaceName "/" * ") hit-cnt " EventCount:int *;
+      let all_106100_unparsed = allLogs
+                | where DeviceEventClassID in ("106100") and isempty(SourceIP)
+                | parse Message with * "access-list " NetworkRuleName " " DeviceAction " " Protocol " " SrcInterfaceName "/" SrcIpAddr "(" SrcPortNumber:int ") -> " DstInterfaceName "/" DstIpAddr "(" DstPortNumber:int ") hit-cnt " EventCount:int * 
+                | extend SimplifiedDeviceAction = case(DeviceAction == "denied", "Deny",
+                                                        "Allow");
+      let remainingLogs = allLogs
+                | where DeviceEventClassID in ("106002", "106012", "106013", "106020");
+      let networkaddressWatchlistData = materialize (_ASIM_GetWatchlistRaw("NetworkAddresses"));
+      let internalInterface = networkaddressWatchlistData | where WatchlistItem.Tags has "Internal" | distinct tostring(WatchlistItem["Range Name"]);
+      let externalInterface = networkaddressWatchlistData | where WatchlistItem.Tags has "External" | distinct tostring(WatchlistItem["Range Name"]);
+      union isfuzzy=true all_106001_alike, all_106010_alike, all_106018, all_106023, all_106023_unparsed, all_106023_41,  all_302013_302015_unparsed, all_302013_302015_parsed, all_302014_302016, all_302020_302021, all_7_series, all_106007, all_106017, all_106100_parsed, all_106100_unparsed, remainingLogs
+                | extend 
+                        EventStartTime = TimeGenerated,
+                        EventEndTime = TimeGenerated,
+                        EventVendor = "Cisco",
+                        EventProduct = "ASA",
+                        EventCount = coalesce(EventCount,toint(1)),
+                        EventType = "NetworkSession",
+                        EventSchema = "NetworkSession",
+                        EventSchemaVersion = "0.2.4",
+                        DvcAction = case(trim(@"\s",SimplifiedDeviceAction) in ("Teardown","Built"), "",
+                                        SimplifiedDeviceAction),
+                        SrcInterfaceName = tolower(SrcInterfaceName),
+                        DstInterfaceName = tolower(SrcInterfaceName)
+                | lookup ProtocolLookup on Protocol
+                | project-rename 
+                        EventProductVersion = DeviceVersion,
+                        EventOriginalType = DeviceEventClassID,
+                        EventOriginalSeverity = OriginalLogSeverity,
+                        DvcOriginalAction = DeviceAction,
+                        EventMessage = Message,
+                        Dvc = Computer
+                | extend
+                        EventSeverity = iff(isempty(DvcAction) or DvcAction == "Allow", "Informational", "Low"),                             
+                        DstPortNumber = coalesce(DstPortNumber, DestinationPort),
```

Comment:
For performance reasons, it is better to avoid those and include the choice of fields in the union branches. This prevents the branches in which he original CEF field is fine from becoming calculated fields.

Reply:
Updated